src/reducers/reducer.js

```diff
@@ -1411,6 +1413,34 @@ export const onNanoContractHistoryFailure = (state, { payload }) => ({
   },
 });
 
+/**
+ * @param {Object} state
+ * @param {{
+ *   payload: {
+ *     ncId: string;
+ *   }
+ * }} action
+ */
+export const onNanoContractHistoryClean = (state, { payload }) => ({
```

Comment:
I added this reducer just to clean history before a new fetch from the beginning.